packages/edit-navigation/src/components/menu-editor/use-navigation-blocks.js

```diff
@@ -1,142 +1,182 @@
 /**
  * External dependencies
  */
-import { groupBy, isEqual, difference } from 'lodash';
+import { groupBy, sortBy, difference } from 'lodash';
 
 /**
  * WordPress dependencies
  */
+import apiFetch from '@wordpress/api-fetch';
 import { createBlock } from '@wordpress/blocks';
-import { useSelect, useDispatch } from '@wordpress/data';
+import { useDispatch, useSelect } from '@wordpress/data';
 import { useState, useRef, useEffect } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 
-function createBlockFromMenuItem( menuItem, innerBlocks = [] ) {
-	return createBlock(
-		'core/navigation-link',
-		{
-			label: menuItem.title.rendered,
-			url: menuItem.url,
-		},
-		innerBlocks
-	);
-}
-
-function createMenuItemAttributesFromBlock( block ) {
-	return {
-		title: block.attributes.label,
-		url: block.attributes.url,
-	};
-}
+/**
+ * Internal dependencies
+ */
+import batchSave from './batch-save';
+import useDebouncedValue from './use-debounced-value';
+import PromiseQueue from './promise-queue';
 
 export default function useNavigationBlocks( menuId ) {
+	const [ deletedMenuItemsIds, setDeletedMenuItemsIds ] = useState( [] );
```

Comment:
This is a hack to work around re-appearing menu items. I'd like to get rid of it and only rely on the data layer in a follow-up PR.

Reply:
Merging the delete part will make it work.